### ISSUE TITLE
Lookup.run can exit and the state can be not done causing an exception in checkDone

### ISSUE THREAD

**Issue by jschewebbn** [dnsjava/dnsjava#137]
The method `Lookup.run` can exit with done still set to false and this causes errors in methods that call it. In my case I have this stack trace using dnsjava 3.2.2. Here the call to `Lookup.run` in `Address.getByName` returned null. However since there isn't an else statement in the if/else inside `if(!done)` of `Lookup.run` it's possible for the method to exit with `done` set to false and then `Lookup.checkDone` throws an exception even though `Lookup.run` has completed.

```
        at org.xbill.DNS.Lookup.checkDone(Lookup.java:621)
        at org.xbill.DNS.Lookup.getResult(Lookup.java:659)
        at org.xbill.DNS.Address.lookupHostName(Address.java:264)
        at org.xbill.DNS.Address.getByName(Address.java:309)
```

**Comment by jschewebbn** [comment]
I believe this shows up when the list of search domains is an array of size zero.

**Comment by jschewebbn** [comment]
A related item, you may want to make it clear which calls to setDefaultSearchDomains accept null and possibly have one separate method to set the value to null explicitly, something like clearDefaultSearchDomains.

**Comment by ibauersachs** [comment]
3.3.0 contains a fix (#118) relating to search domains and Lookup.run. Have you tried that?

**Comment by jschewebbn** [comment]
It looks like using version 3.3.0 fixes the problem when the search domains is an array of size zero.
